### PR TITLE
ADEN-10452 Adding UAP template to New Mobile Experience

### PR DESCRIPTION
**Pull request by dmnsobczak**

### CODE REVIEW

File: platforms/ucp-mobile/setup/dynamic-slots/ucp-mobile-dynamic-slots.setup.ts [hunk 1]
```diff
@@ -93,6 +94,13 @@ export class UcpMobileDynamicSlotsSetup implements DiProcess {
 		fillerService.register(new PorvataFiller());
 	}
 
+	private registerTopLeaderboardCodePriority(): void {
+		// ToDo: check if pageType detection works fine on fandommobile
```
Comment (nandy-andy) [inline platforms/ucp-mobile/setup/dynamic-slots/ucp-mobile-dynamic-slots.setup.ts:98]: Do we want to check it within ADEN-10452 or create a follow-up?

File: platforms/ucp-mobile/setup/dynamic-slots/ucp-mobile-slots-definition-repository.ts [hunk 2]
```diff
@@ -58,8 +58,8 @@ export class UcpMobileSlotsDefinitionRepository {
 
 	private isTopLeaderboardApplicable(): boolean {
 		const hasFeaturedVideo = context.get('custom.hasFeaturedVideo');
-		const isHome = context.get('custom.pageType') === 'home';
-		const isSearch = context.get('custom.pageType') === 'search';
+		const isHome = context.get('wiki.opts.pageType') === 'home';
+		const isSearch = context.get('wiki.opts.pageType') === 'search';
```
Comment (nandy-andy) [inline platforms/ucp-mobile/setup/dynamic-slots/ucp-mobile-slots-definition-repository.ts:62]: What do you think of making `isHome` and `isSearch` properties of `UcpMobileSlotsDefinitionRepository`, assigning its values in the constructor and then checking the properties here and in lines 94 and 98?

File: platforms/ucp-mobile/templates/bfab-template.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+				BfabBootstrapHandler,
+				VideoBootstrapHandler,
+				AdvertisementLabelHandler,
+				DebugTransitionHandler,
+			],
+			impact: [
+				SlotSizeImpactHandler,
+				VideoSizeImpactHandler,
+				VideoCompletedHandler,
+				VideoLearnMoreHandler,
```
Comment (Loczi94) [inline platforms/ucp-mobile/templates/bfab-template.ts:38]: To discuss.

File: platforms/ucp-mobile/setup/dynamic-slots/ucp-mobile-dynamic-slots.setup.ts [hunk 4]
```diff
@@ -93,6 +94,13 @@ export class UcpMobileDynamicSlotsSetup implements DiProcess {
...
+		if (!context.get('custom.hasFeaturedVideo') && context.get('wiki.opts.pageType') !== 'search') {
```
[the remark this answers is not in the document]
Comment (Loczi94) [reply platforms/ucp-mobile/setup/dynamic-slots/ucp-mobile-dynamic-slots.setup.ts:99]: I was wrong - it's in UcpTargetingSetup. 🙃

**Review by Loczi94 — APPROVED** [review]

File: platforms/ucp-mobile/setup/dynamic-slots/ucp-mobile-dynamic-slots.setup.ts (hunk 1 above)
Comment (dmnsobczak) [reply platforms/ucp-mobile/setup/dynamic-slots/ucp-mobile-dynamic-slots.setup.ts:98]: Follow up for all gathered issues, let's keep the scope of this ticket only for UAP :)

File: platforms/ucp-mobile/setup/dynamic-slots/ucp-mobile-slots-definition-repository.ts (hunk 2 above)
Comment (dmnsobczak) [reply platforms/ucp-mobile/setup/dynamic-slots/ucp-mobile-slots-definition-repository.ts:62]: I will copy this suggestion to the follow up ticket focused on `pageType` variable, let's test only UAP before demo :)

**Pull request merged**